utils/s2n_random.c

```diff
@@ -390,7 +390,20 @@ int s2n_get_rdrand_data(struct s2n_blob *out)
             :"cc");
 #endif /* defined(__i386__) */
 
-            if (success) {
+            /* Some AMD CPUs will find that RDRAND "sticks" on all 1s but still reports success.
```

Comment:
If we compile this code with i386 as the target, will we use the code that's between lines 360 and 378? If we do that, are there any cases where the AMD suspend bug might cause the first result (the low bytes) to be all ones or all zeroes, but not the second?  Or it might return all-ones the first time, and all-zeroes the second time? A case like this would evade our checks, but be biased. 

From everything I read though it looks like that's not how the bug works.

Reply:
Good catch on the two half check - I can certainly add a check to clear succes in the i386 if either half is all 1s or all 0s to avoid the bias.

Reply:
My understanding of the AMD bug is that if it is stuck it will _always_ return all `1`s. Thus if only one of the two 32-bit returns is all `1`s, then (by definition) it isn't stuck.

However, as noted, for some very old processors all `0`s is the signal for bad output. Thus even if only one 32-bit output is all `0`s, it should be rejected.